django-backend/fecfiler/memo_text/models.py

```diff
@@ -15,10 +19,55 @@ class MemoText(SoftDeleteModel, CommitteeOwnedModel, ReportMixin):
     )
     rec_type = models.TextField(null=True, blank=True)
     filer_committee_id_number = models.TextField(null=True, blank=True)
+    is_report_level_memo = models.BooleanField(null=False, blank=False, default=True)
     transaction_id_number = models.TextField(null=True, blank=True)
+    transaction_uuid = models.TextField(null=True, blank=True)
     back_reference_tran_id_number = models.TextField(null=True, blank=True)
     back_reference_sched_form_name = models.TextField(null=True, blank=True)
     text4000 = models.TextField(null=True, blank=True)
 
     class Meta:
         db_table = "memo_text"
+
+    # This is intended to be useable without instantiating a transaction object
+    @staticmethod
+    def check_for_uid_conflicts(uid, report_id):  # noqa
+        return (
+            len(MemoText.objects.filter(transaction_id_number=uid, report_id=report_id))
+            > 0
+        )
+
+    def generate_report_id(self):
+        report_memos = MemoText.objects.filter(
+            report_id=self.report_id, transaction_uuid=None
+        )
+        return "REPORT_MEMO_TEXT" + str(len(report_memos) + 1)
+
+    def generate_uid(self):
+        unique_id = uuid.uuid4()
+        hex_id = unique_id.hex.upper()
+        # Take 20 characters from the end, skipping over the 20th char from the right,
+        # which is the version number (uuid4 -> "4")
+        return hex_id[-21] + hex_id[-19:]
```

Comment:
This method, generate_uid(),  is the exact same as the one of same name in the scha_transactions model. Let's put it in the new "shared" app that contains the common methods across apps/models.

Reply:
Great call; taken care of